### PR TITLE
PipelineLayout Invalidates Pipelines

### PR DESCRIPTION
**Pull request by purvisa-at-google-com**

### CODE REVIEW

File: gapis/api/vulkan/frame_loop.go [hunk 1]
```diff
@@ -1326,6 +1330,61 @@ func (f *frameLoop) resetPipelineLayouts(ctx context.Context, stateBuilder *stat
...
+	// The shadow state for PipelineLayouts does not contain reference to the ComputePipelines they are used in. So we have to loop around finding the story.
+	for _, computePipelineObject := range GetState(f.loopStartState).ComputePipelines().All() {
+		pipelineLayout := computePipelineObject.PipelineLayout()
+		if _, ok := f.pipelineLayoutToCreate[pipelineLayout.VulkanHandle()]; ok {
+			f.pipelineLayoutToDestroy[pipelineLayout.VulkanHandle()] = true
+			f.pipelineLayoutToCreate[pipelineLayout.VulkanHandle()] = true
+		}
+	}
+
+	// The shadow state for PipelineLayouts does not contain reference to the GraphicsPipelines they are used in. So we have to loop around finding the story
```
Comment (hevrard) [inline gapis/api/vulkan/frame_loop.go]: Style: add a dot at the end of the comment.

File: gapis/api/vulkan/frame_loop.go [hunk 2]
```diff
@@ -1326,6 +1330,61 @@ func (f *frameLoop) resetPipelineLayouts(ctx context.Context, stateBuilder *stat
...
+	for _, graphicsPipelineObject := range GetState(f.loopStartState).GraphicsPipelines().All() {
+		pipelineLayout := graphicsPipelineObject.Layout()
```
Comment (hevrard) [inline gapis/api/vulkan/frame_loop.go]: On a compute pipeline you call `.PipelineLayout()`, whereas on a graphics pipeline you call `.Layout()`, I would have expected to see `.PipelineLayout()` in both cases. Can you confirm this is doing the correct thing here?

File: gapis/api/vulkan/frame_loop.go [hunk 3]
```diff
@@ -1326,6 +1330,61 @@ func (f *frameLoop) resetPipelineLayouts(ctx context.Context, stateBuilder *stat
...
+		} else if isGraphicsPipeline {
+			stateBuilder.write(stateBuilder.cb.VkDestroyPipeline(graphicsPipeline.Device(), graphicsPipeline.VulkanHandle(), memory.Nullptr))
+		} else {
+			log.F(ctx, true, "FrameLooping: resetPipelines(): Unknown pipeline type")
+		}
+	}
+
+	// For every ComputePipeline that we need to create at the end of the loop...
+	for toCreate := range f.computePipelineToCreate {
+		// Write the commands needed to recreate the destroyed object
```
Comment (hevrard) [inline gapis/api/vulkan/frame_loop.go]: Style: missing dots at the end of comment.

File: gapis/api/vulkan/frame_loop.go [hunk 4]
```diff
@@ -1326,6 +1330,61 @@ func (f *frameLoop) resetPipelineLayouts(ctx context.Context, stateBuilder *stat
...
+		stateBuilder.createComputePipeline(GetState(f.loopStartState).computePipelines.Get(toCreate))
+	}
+
+	// For every GraphicsPipeline that we need to create at the end of the loop...
+	for toCreate := range f.graphicsPipelineToCreate {
+		// Write the commands needed to recreate the destroyed object
```
Comment (hevrard) [inline gapis/api/vulkan/frame_loop.go]: Style: missing dots at the end of comment.

File: gapis/api/vulkan/frame_loop.go (hunk 2 above)
Comment (purvisa-at-google-com) [reply gapis/api/vulkan/frame_loop.go]: Yup. The names in the API files are not always consistent sadly.

**Pull request merged**